apps/activity_logger/lib/activity_logger/activity_log.ex

```diff
@@ -0,0 +1,190 @@
+defmodule ActivityLogger.ActivityLog do
+  @moduledoc """
+  Ecto Schema representing activity_logs.
+  """
+  use Ecto.Schema
+  use Utils.Types.ExternalID
+  import Ecto.{Changeset, Query}
+  alias Ecto.{Changeset, UUID}
+
+  alias ActivityLogger.{
+    ActivityLog,
+    Repo
+  }
+
+  @primary_key {:uuid, UUID, autogenerate: true}
+  @primary_key {:uuid, UUID, autogenerate: true}
+
+  schema "activity_log" do
+    external_id(prefix: "adt_")
+
+    field(:action, :string)
+
+    field(:target_type, :string)
+    field(:target_uuid, UUID)
+    field(:target_changes, :map)
+    field(:target_encrypted_changes, ActivityLogger.Encrypted.Map, default: %{})
+
+    field(:originator_uuid, UUID)
+    field(:originator_type, :string)
+
+    field(:metadata, :map, default: %{})
+
+    field(:inserted_at, :naive_datetime)
+  end
+
+  defp changeset(changeset, attrs) do
+    changeset
+    |> cast(attrs, [
+      :action,
+      :target_type,
+      :target_uuid,
+      :target_changes,
+      :target_encrypted_changes,
+      :originator_uuid,
+      :originator_type,
+      :metadata,
+      :inserted_at
+    ])
+    |> validate_required([
+      :action,
+      :target_type,
+      :target_uuid,
+      :target_changes,
+      :originator_uuid,
+      :originator_type,
+      :inserted_at
+    ])
+  end
+
+  @spec get_schema(String.t()) :: Atom.t()
```

Comment:
There's no `Atom.t()`.  Use`atom()` instead.

Also applies to the rest of the PR.